src/components/pages/BulkAddServices.tsx

```diff
@@ -0,0 +1,125 @@
+
+import { InactiveChip } from './Project';
+
+import useSafeParams from '@/hooks/useSafeParams';
+import BulkAdd from '@/modules/bulk/components/BulkAdd';
+import ProjectEnrollmentsTable from '@/modules/inventory/components/ProjectEnrollmentsTable';
+import ProjectLayout from '@/modules/inventory/components/ProjectLayout';
+import { useProjectCrumbs } from '@/modules/inventory/components/useProjectCrumbs';
+import {
+  AddServiceToEnrollmentDocument,
+  AddServiceToEnrollmentMutation,
+  AddServiceToEnrollmentMutationVariables,
+  EnrollmentFieldsFragment,
+} from '@/types/gqlTypes';
+
+const BulkAddServices = () => {
+  const { projectId } = useSafeParams() as {
+    projectId: string;
+  };
+  const title = 'Record Services';
+  const [crumbs, crumbsLoading, project] = useProjectCrumbs(title);
+  const [workingEnrollmentId, setWorkingEnrollmentId] = useState<
+    string | undefined
+  >();
+  const [enrollmentsAdded, setEnrollmentsAdded] = useState<string[]>([]);
+
+  if (crumbsLoading) return <Loading />;
+  if (!crumbs || !project) throw Error('Project not found');
+
+  return (
+    <ProjectLayout crumbs={crumbs}>
+      <BulkAdd<
+        EnrollmentFieldsFragment,
+        AddServiceToEnrollmentMutation,
+        AddServiceToEnrollmentMutationVariables
+      >
+        mutationDocument={AddServiceToEnrollmentDocument}
+        renderList={(items, { onSelect, mutationLoading }) => (
+          <ProjectEnrollmentsTable
+            noLinks
+            projectId={projectId}
+            additionalColumns={[
+              ...items.map(
+                (item) =>
+                  ({
+                    header: item.label,
+                    render: (enrollment) =>
+                      item.getNode(enrollment, {
+                        disabled: enrollmentsAdded.includes(enrollment.id),
+                      }),
+                  } as ColumnDef<EnrollmentFieldsFragment>)
+              ),
+              {
+                header: '',
+                render: (enrollment) => (
+                  <Box sx={{ textAlign: 'right' }}>
+                    <Button
+                      color='secondary'
+                      onClick={() => {
+                        onSelect(enrollment);
+                        setWorkingEnrollmentId(enrollment.id);
+                      }}
+                      disabled={enrollmentsAdded.includes(enrollment.id)}
+                      startIcon={
+                        mutationLoading &&
+                        workingEnrollmentId === enrollment.id ? (
+                          <CircularProgress color='inherit' size={15} />
+                        ) : enrollmentsAdded.includes(enrollment.id) ? (
+                          <CheckIcon />
+                        ) : undefined
+                      }
+                    >
+                      {enrollmentsAdded.includes(enrollment.id)
+                        ? 'Assigned'
+                        : 'Assign'}
+                    </Button>
```

Comment:
I'm seeing a bug that is allowing me to create duplicate records if I assign services to multiple clients quickly without waiting for each one to finish loading. The loading spinner disappears because we only have one "workingEnrollmentId" even though several can be "working" at the same time. I'm able to go click on a previously assigned service again, and it creates a duplicate service.

A quick way to fix this might be to change `workingEnrollmentId` state to a list ("loadingEnrollmentIds") so you can store all of them, and remove them one by one as mutations succeed. Finally, you want to make the button `disabled` if the mutation is currently loading.

The way I handled this for adding household members was to pull the mutation and loading state into the button itself (`AddToHouseholdButton`) which keeps track of it's own loading/success/error status, and calls `onSuccess` hook when it has successfully completed. I think that approach could work well here too, and that could potentially be refactored into a generic component MutationButton. I'll keep that in mind, but for now, if the above suggestion works to fix this bug, I'd go with that.